leaf-ui/js/object/evoObjects.js

```diff
@@ -141,42 +103,41 @@ class EVO {
      */
     static setCurTimePoint(newTimePoint, analysisResult) {
         EVO.curTimePoint = newTimePoint;
-        IntentionColoring.refresh(analysisResult);
+        EVO.refresh(analysisResult);
     }
 
     /**
      * Runs after any event that may change visualization, such as setting a sat value, changing slider option, or selecting a time point
      */
     static refresh(analysisResult) {
-        switch(this.sliderOption) {
-            case '1':
-            case '2':
-            case '3':
-                if (analysisResult !== undefined){
-                    if (analysisResult.get('selected')){
-                        EVO.colorIntentionsAnalysis(analysisResult);
-                    }
-                    else {
-                        EVO.colorIntentionsModeling();
-                    }
+        if (this.sliderOption > 0) {
+            if (analysisResult !== undefined){
+                if (analysisResult.get('selected')){
+                    EVO.colorIntentionsAnalysis(analysisResult);
                 }
-                else{
+                else {
                     EVO.colorIntentionsModeling();
                 }
-                EVO.changeIntentionsText(analysisResult);
-                break;
-            default://colorVis off
-                EVO.returnAllColors(graph.getElements(), paper);
+            }
+            else{
+                EVO.colorIntentionsModeling();
+            }
+            EVO.changeIntentionsText(analysisResult);
+        } else {
+            EVO.returnAllColors(graph.getElements(), paper);
+            EVO.revertIntentionsText(graph.getElements(), paper);
                 EVO.revertIntentionsText(graph.getElements(), paper);    
-                    break;
+            EVO.revertIntentionsText(graph.getElements(), paper);
+                EVO.revertIntentionsText(graph.getElements(), paper);    
+            EVO.revertIntentionsText(graph.getElements(), paper);
```

Comment:
Why is `EVO.revertIntentionsText(graph.getElements(), paper);` repeated four times?